Research hub landing page

danielfmiranda opened the pull request:
Closes #8314
Related PRs/issues #

tbrlpld commented on network-api/networkapi/wagtailpages/pagemodels/research_hub/landing_page.py:
```diff
@@ -1,7 +1,16 @@
+    FieldPanel, InlinePanel
+)
+
+from ...utils import (
```
**Suggestion:** I think an absolute include would be clearer in this case, because there are many `utils` modules.

danielfmiranda replied:
Updated!

tbrlpld commented on network-api/networkapi/wagtailpages/pagemodels/research_hub/landing_page.py:
```diff
@@ -11,8 +20,35 @@ class ResearchLandingPage(foundation_metadata.FoundationMetadataPageMixin, wagta
+    )
+
+    content_panels = [
+        FieldPanel(
+            'title',
+            classname='full title',
+            widget=TitleWidget(attrs={"class": "max-length-warning", "data-max-length": 60})
+        ),
```
**Suggestion:** To avoid having to redefine the `title` field this elaborately, it is very common in Wagtail to do 
```python
content_panels = wagtail_models.Page.content_panels + [
    ...  # list the custom panels you want to use. 
]
```

danielfmiranda replied:
Updated!

tbrlpld commented on network-api/networkapi/wagtailpages/pagemodels/research_hub/landing_page.py:
```diff
@@ -11,8 +20,35 @@ class ResearchLandingPage(foundation_metadata.FoundationMetadataPageMixin, wagta
         context = super().get_context(request)
         ResearchLibraryPage = apps.get_model("wagtailpages", "ResearchLibraryPage")
         context['library_page'] = ResearchLibraryPage.objects.first()
+        context['latest_research_detail_pages'] = self.get_recent_research_pages
```
**Suggestion:** For consistency, I think it would be nice if we stick to either "latest" or "recent", but not mix them. E.g. use `get_latest_research_detail_pages` for the method to match the context variable.

danielfmiranda replied:
Updated!

tbrlpld commented on network-api/networkapi/wagtailpages/pagemodels/research_hub/landing_page.py:
```diff
@@ -11,8 +20,35 @@ class ResearchLandingPage(foundation_metadata.FoundationMetadataPageMixin, wagta
+        context['latest_research_detail_pages'] = self.get_recent_research_pages
         return context
+
+    def get_recent_research_pages(self, *, search='', sort=None):
```
**Issue:** We should remove the unused arguments. These are suggesting functionality we are not providing.

danielfmiranda replied:
Updated!

tbrlpld commented on network-api/networkapi/wagtailpages/pagemodels/research_hub/relations.py:
```diff
@@ -65,3 +65,23 @@ class ResearchDetailPageResearchTopicRelation(wagtail_models.TranslatableMixin,
+
+
+class ResearchLandingPageFeaturedResearchTopicRelation(wagtail_models.TranslatableMixin, wagtail_models.Orderable):
+    research_detail_page = cluster_fields.ParentalKey(
```
**Issue:** This attribute needs to be renamed to match the actually linked page type. 
```suggestion
    research_landing_page = cluster_fields.ParentalKey(
```

danielfmiranda replied:
Updated!

tbrlpld reviewed (changes requested):
@danielfmiranda I think are a few quick fixes that would be helpful to make and some suggestions for your consideration.

tbrlpld commented on network-api/networkapi/wagtailpages/templates/wagtailpages/research_landing_page.html:
```diff
@@ -2,10 +2,67 @@
+  <ul>
+    {% for featured_topic in page.featured_topics.all %}
+    <li>
+      <h3><a href="{% pageurl library_page %}?topic={{featured_topic.id}}">{{ featured_topic.research_topic.name }}</a></h3>
```
**Issue:** I don't think this works. It should be `?topic={{ featured_topic.research_topic.id }}`. Currently it uses the `id` of the through models and not of the topic itself.

danielfmiranda replied:
Updated!

danielfmiranda commented:
Hi @tbrlpld thanks for the great finds! I have implemented your feedback and re-requested review, however, it appears that the tests for the library page are now failing after I added some featured topics to the landing page factory. I was wondering if we could work through that issue together so I can get more familiar with tests? Thanks!

tbrlpld reviewed (approved):
Thanks for incorporating those suggestions @danielfmiranda  Looking good 👍
